Xamarin.Forms.Platform.MacOS/Controls/FormsScrollView.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using AppKit;
+using Foundation;
+using ObjCRuntime;
+using PointF = CoreGraphics.CGPoint;
+
+namespace Xamarin.Forms.Platform.MacOS
+{
+	internal class FormsScrollView : NSScrollView
+	{
+		bool _shouldHandleCustomScrolling = true;
+
+		public EventHandler<FormsScrollViewScrollChangedEventArgs> ScrollChanged;
+
+		public FormsScrollView(bool handleCustomScrolling = true)
+		{
+			_shouldHandleCustomScrolling = handleCustomScrolling;
+
+			ContentView.PostsBoundsChangedNotifications = true;
+
+			NSNotificationCenter.DefaultCenter.AddObserver(this, new Selector(nameof(UpdateScrollPosition)), NSView.BoundsChangedNotification, ContentView);
+		}
+
+		public nfloat CustompageScroll
+		{
+			get;
+			set;
+		}
+
+		public override async void ScrollWheel(NSEvent theEvent)
```

Comment:
Note that it **is** still called a ScrollWheel event by Apple in [Mouse Events](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/EventOverview/EventObjectsTypes/EventObjectsTypes.html) especially as there is a virtual scrollwheel on the MagicMouse (and my Windows mice still have them).

However, yes you're right that the OS routes two-finger scrolling gestures on the Trackpack through this event.

As WPF also has a _Wheel_ (Mouse.MouseWheel) I'm in favour of keeping it as named - it keeps the event more aligned with the target platform's framework, which helps people translating code.